### PR TITLE
added immersive and regular modals

### PR DESCRIPTION
**Pull request by ZeMunchkin**
Trello: https://trello.com/c/5arVBzoV

- Added Modal and ImmersiveModal (and necessary utils) to radiance library 
- Added MenuIcon to icon library

### CODE REVIEW

File: docs/immersiveModal.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  render() {
+    const { defaultIsOpen, headerIsOpen } = this.state;
+
+    return (
+      <div>
+        <Button onClick={this.onOpenDefaultModal}>Open ImmersiveModal</Button>
+        {defaultIsOpen && (
+          <ImmersiveModal onClose={this.onClose}>
+            <ImmersiveModal.Title>This is styled with ImmersiveModalTitle</ImmersiveModal.Title>
```
Comment (snags88) [inline docs/immersiveModal.md]: probably want to update the text to say "This is styled with `ImmersiveModal.Title`". Similar comment for the rest of the file

File: docs/immersiveModal.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  }
+}
+```
+
+<!-- STORY -->
+
+### Proptypes
+| prop     | propType           | required | default | description                                                                                                                  |
+|----------|--------------------|----------|---------|------------------------------------------------------------------------------------------------------------------------------|
+| onClose | func | no | () => {} | function is that executed when either the close icon is clicked or any part of the page outside the modal is clicked |
```
Comment (snags88) [inline docs/immersiveModal.md]: function that is*

File: docs/modal.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  }
+}
+```
+
+<!-- STORY -->
+
+### Proptypes
+| prop     | propType           | required | default | description                                                                                                                  |
+|----------|--------------------|----------|---------|------------------------------------------------------------------------------------------------------------------------------|
+| onClose | func | no | () => {} | Function is that executed when the close icon is clicked |
```
Comment (snags88) [inline docs/modal.md]: function that is*

File: src/constants/keycodes/index.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import throwOnUndefinedProperty from '../../utils/throwOnUndefinedProperty';
+
+const keycodes = throwOnUndefinedProperty({
+  backspace: 8,
+  enter: 13,
+  escape: 27,
+  space: 32,
+  h: 104, // used for hugpacks
```
Comment (snags88) [inline src/constants/keycodes/index.js]: We can strip out the ones we don't need. Probably only need `escape`

File: src/shared-components/immersiveModal/index.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  static Body = Body;
+
+  static Footer = Footer;
+
+  constructor(props) {
+    super(props);
+
+    this.htmlNode = document.querySelector('html');
+    this.domNode =
+      document.querySelector('#reactPortalSection') || document.body;
```
Comment (snags88) [inline src/shared-components/immersiveModal/index.js]: just curious what happens if query selector is null. It seems from the code that it's not a hard fast requirement to have a node for the react portal

File: src/shared-components/immersiveModal/index.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  }
+
+  componentDidMount() {
+    this.htmlNode.classList.add('no-scroll');
```
Comment (snags88) [inline src/shared-components/immersiveModal/index.js]: just wanted to check if `no-scroll` class was part of the inject global styles

File: src/shared-components/modal/index.js [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+  static Body = Body;
+
+  static Footer = Footer;
+
+  state = {
+    isVisible: false,
+    parentNode: document.querySelector('#reactPortalSection') || document.body,
+  };
+
+  componentDidMount() {
```
Comment (snags88) [inline src/shared-components/modal/index.js]: can we add these listeners to the immersive modal?

File: src/shared-components/modal/index.js [hunk 8]
```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ReactModal from 'react-modal';
```
Comment (snags88) [inline src/shared-components/modal/index.js]: as a follow up (not high priority), i'd like for us to refactor this modal to not use ReactModal. I don't think we need it and we're doing the immersive modal without it.

File: src/shared-components/immersiveModal/index.js (hunk 6 above)
Comment (ZeMunchkin) [reply src/shared-components/immersiveModal/index.js]: Confirmed -- just checked in radiance, gatsby, and main app and it's the same in all three.

File: src/shared-components/modal/index.js (hunk 8 above)
Comment (ZeMunchkin) [reply src/shared-components/modal/index.js]: Created an issue for follow up: https://github.com/PocketDerm/radiance-ui/issues/40#issue-403271409

**Comment by ZeMunchkin** [comment]
@snags88 Updated!

File: src/shared-components/immersiveModal/index.js (hunk 6 above)
Comment (snags88) [reply src/shared-components/immersiveModal/index.js]: 🙌

**Pull request merged**